packages/shared/src/components/modals/SlackIntegrationModal.tsx

```diff
@@ -0,0 +1,351 @@
+import React, { ReactElement, useCallback, useState } from 'react';
+import { useMutation, useQueryClient } from '@tanstack/react-query';
+import { Modal, ModalProps } from './common/Modal';
+import {
+  UserIntegration,
+  UserIntegrationType,
+} from '../../graphql/integrations';
+import { Dropdown } from '../fields/Dropdown';
+import { ButtonSize, ButtonVariant } from '../buttons/common';
+import { Button } from '../buttons/Button';
+import { useToastNotification, useViewSize, ViewSize } from '../../hooks';
+import { Source, SourceType } from '../../graphql/sources';
+import { ModalClose } from './common/ModalClose';
+import { SourceAvatar } from '../profile/source';
+import { ProfileImageSize } from '../ProfilePicture';
+import {
+  Typography,
+  TypographyColor,
+  TypographyTag,
+  TypographyType,
+} from '../typography/Typography';
+import { PlusIcon, SlackIcon } from '../icons';
+import { useSlack } from '../../hooks/integrations/slack/useSlack';
+import { useIntegrations } from '../../hooks/integrations/useIntegrations';
+import { useSourceIntegration } from '../../hooks/integrations/useSourceIntegration';
+import { useSlackChannels } from '../../hooks/integrations/slack/useSlackChannels';
+import { useIntegration } from '../../hooks/integrations/useIntegration';
+import { generateQueryKey, RequestKey } from '../../lib/query';
+import { useAuthContext } from '../../contexts/AuthContext';
+import { labels } from '../../lib';
+import { Image } from '../image/Image';
+import { useThemedAsset } from '../../hooks/utils';
+import { slackIntegration } from '../../lib/constants';
+import { LazyModal } from './common/types';
+import { Loader } from '../Loader';
+
+export type SlackIntegrationModalProps = Omit<ModalProps, 'children'> & {
+  source: Pick<Source, 'id' | 'handle' | 'type' | 'image' | 'name'>;
+};
+
+const SlackIntegrationModal = ({
+  source,
+  ...props
+}: SlackIntegrationModalProps): ReactElement => {
+  const isMobile = useViewSize(ViewSize.MobileL);
+  const { removeSourceIntegration } = useIntegration();
+  const queryClient = useQueryClient();
+  const { user } = useAuthContext();
+  const { displayToast } = useToastNotification();
+
+  const [state, setState] = useState<{
+    userIntegration?: UserIntegration;
+    channelId?: string;
+  }>({});
+
+  const { data: sourceIntegration } = useSourceIntegration({
+    userIntegrationType: UserIntegrationType.Slack,
+    sourceId: source.id,
+  });
+
+  const { data: slackIntegrations, isLoading: isLoadingIntegrations } =
+    useIntegrations({
+      queryOptions: {
+        select: useCallback((data) => {
```

Comment:
This is just a future proofing for different integrations?
Feels like a lot of overhead now, but maybe it's fine.

Just wondering if `useIntegrations` is still valid, but then have a `useSlackIntegration` which does abstraction on top is better?

Reply:
its valid yes because `useSlackIntegration` checks single integration for source